NuRadioMC/simulation/simulation.py

```diff
@@ -1,85 +1,1031 @@
 import NuRadioReco.modules.channelResampler
-import NuRadioReco.detector.detector as detector
+import NuRadioReco.modules.triggerTimeAdjuster
+from NuRadioReco.detector import detector
 import NuRadioReco.framework.sim_station
 import NuRadioReco.framework.electric_field
 import NuRadioReco.framework.particle
 import NuRadioReco.framework.event
 import NuRadioReco.framework.sim_emitter
 from NuRadioReco.detector import antennapattern
-from NuRadioReco.utilities import geometryUtilities as geo_utl
-from NuRadioReco.framework.parameters import channelParameters as chp
 from NuRadioReco.framework.parameters import electricFieldParameters as efp
 from NuRadioReco.framework.parameters import showerParameters as shp
+from NuRadioReco.framework.parameters import channelParameters as chp
 # parameters describing simulated Monte Carlo particles
 from NuRadioReco.framework.parameters import particleParameters as simp
 from NuRadioReco.framework.parameters import emitterParameters as ep
 # parameters set in the event generator
 from NuRadioReco.framework.parameters import generatorAttributes as genattrs
-import datetime
-import logging
-import warnings
-from six import iteritems
-import yaml
-import os
-import collections
-from NuRadioMC.utilities.Veff import remove_duplicate_triggers
-import logging
-from NuRadioReco.utilities.logging import LOGGING_STATUS, setup_logger
+import NuRadioMC.simulation.time_logger
 
 logger = setup_logger("NuRadioMC.simulation")
 
-
-def pretty_time_delta(seconds):
-    seconds = int(seconds)
-    days, seconds = divmod(seconds, 86400)
-    hours, seconds = divmod(seconds, 3600)
-    minutes, seconds = divmod(seconds, 60)
-    if days > 0:
-        return '%dd%dh%dm%ds' % (days, hours, minutes, seconds)
-    elif hours > 0:
-        return '%dh%dm%ds' % (hours, minutes, seconds)
-    elif minutes > 0:
-        return '%dm%ds' % (minutes, seconds)
-    else:
-        return '%ds' % (seconds,)
-
+# initialize a few NuRadioReco modules
+# TODO: Is this the best way to do it? Better to initialize them on demand
+channelAddCableDelay = NuRadioReco.modules.channelAddCableDelay.channelAddCableDelay()
+electricFieldResampler = NuRadioReco.modules.electricFieldResampler.electricFieldResampler()
+efieldToVoltageConverterPerEfield = NuRadioReco.modules.efieldToVoltageConverterPerEfield.efieldToVoltageConverterPerEfield()
+efieldToVoltageConverter = NuRadioReco.modules.efieldToVoltageConverter.efieldToVoltageConverter()
+channelSignalReconstructor = NuRadioReco.modules.channelSignalReconstructor.channelSignalReconstructor()
+channelResampler = NuRadioReco.modules.channelResampler.channelResampler()
+channelGenericNoiseAdder = NuRadioReco.modules.channelGenericNoiseAdder.channelGenericNoiseAdder()
+channelResampler = NuRadioReco.modules.channelResampler.channelResampler()
+eventWriter = NuRadioReco.modules.io.eventWriter.eventWriter()
+triggerTimeAdjuster = NuRadioReco.modules.triggerTimeAdjuster.triggerTimeAdjuster()
```

Comment:
Personally I am not a fan of calling the object the same as the module/class. If we would change that we could also simplify the module imports.

Reply:
what is your suggestion?

Reply:
same names but in snake_case? `channelAddCableDelay` -> `channel_add_cable_delay` or something similar like `channel_cable_delay_adder`

Reply:
I resolve this for now. This is a general question that is not directly related to the refactor branch. This can be changed everywhere in a new PR.

Reply:
Than it gets never done.